### PR TITLE
[FIX] POS Round Three

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[FIX](https://bcdevex.atlassian.net/browse/FIX)

Objective: 

- Found an issue, round three heuristic rules were not being properly applied
- negative amounts (between payments and deposits) also resolved
sample report: 
(you can clearly see where a more generalized heuristics round to match many to many across multiple days would increase the match % - for round three we are matching one to one, hence the exceptions (see Campbell River for example) 

[reconciliation_report_2023-09-12-2023-09-30.xlsx](https://github.com/bcgov/PaymentCommonComponent/files/12822963/reconciliation_report_2023-09-12-2023-09-30.xlsx)

### CODE REVIEW

File: apps/backend/src/lambdas/reconcile.ts [hunk 1]
```diff
@@ -32,7 +37,7 @@ export const handler = async (event: SNSEvent, _context?: Context) => {
   const notificationService = app.get(NotificationService);
   const snsService = app.get(SnsManagerService);
   const reportingService = app.get(ReportingService);
-  const numDaysToReconcile = 31;
+  const numDaysToReconcile = 14;
```
Comment (fw-noel) [inline apps/backend/src/lambdas/reconcile.ts:40]: I understand this difference probably isn't meaningful in the grand scheme, but can I ask **why** this is a needed change?

File: apps/backend/src/lambdas/reconcile.ts [hunk 2]
```diff
@@ -219,11 +142,139 @@ export const handler = async (event: SNSEvent, _context?: Context) => {
...
+      appLogger
+    );
     reportEnabled && (await generateReport());
     isLocal && (await showConsoleReport());
   } catch (err) {
     appLogger.error(err);
   }
 };
+
+const findPosExceptions = async (
```
Comment (fw-noel) [inline apps/backend/src/lambdas/reconcile.ts:177]: Comments above each of these with expected outcomes please!

**Review by fw-noel — COMMENTED** [review]
Do we have sufficient test coverage for `reconcilePos` and `reconcileCash`?

File: apps/backend/src/lambdas/reconcile.ts (hunk 1 above)
Comment (chelsea-EYDS) [reply apps/backend/src/lambdas/reconcile.ts:40]: (I do think we should decrease the no. of days we're reconciling for but will need the team/Aiden to agree :) )

**Comment by chelsea-EYDS** [comment]
> Do we have sufficient test coverage for `reconcilePos` and `reconcileCash`? If not, would love that in a different PR

These are the same functions just moved outside of the lambda helper for readability so I have not changed or updated any of the tests, but I do not think we have any tests specifically for the lambda. I will make a tech debt ticket to test these :)

**Review by fw-noel — APPROVED** [review]

**Pull request merged**